src/generic/MapWithInverse.jl

```diff
@@ -16,8 +16,8 @@ export map_with_preimage_from_func, map_with_section_from_func,
 
 map_with_section(f::Map{D, C}, g::Map{C, D}) where {D, C} = MapWithSection(f, g)
 
-domain(f::MapWithSection{D, C}) where {D, C} = get_field(f.map, :domain)::D
-codomain(f::MapWithSection{D, C}) where {D, C} = get_field(f.map, :codomain)::C
+domain(f::MapWithSection{D, C}) where {D, C} = domain(f.map::D
```

Comment:
I think there is at least a parenthesis or two missing.

In any event I will want to see the green before merging.

Reply:
Err... WTF?!? OK, I swear I had this running and passing tests locally. No idea how I managed to screw that up :-(. Anyway, hopefully fixed now. And of course it must be green before even be considered for merging, 100% agreed.

Reply:
Thanks.

Reply:
Tests are green now.